src/libstd/sys/unix/thread.rs

```diff
@@ -71,7 +71,7 @@ impl Thread {
         };
 
         extern fn thread_start(main: *mut libc::c_void) -> *mut libc::c_void {
-            unsafe { start_thread(main); }
+            unsafe { start_thread(main, pthread_self() as u32); }
```

Comment:
On Windows what I want is some sort of OS provided ID. Either the thread `HANDLE` or the thread ID which is a `DWORD`. Anything else would be useless to me as I wouldn't be able to use it in OS functions to manipulate that thread like queuing APCs to it.